src/Std/Internal/Async/Signal.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+module
+
+prelude
+public import Std.Time
+public import Std.Internal.UV.Signal
+public import Std.Internal.Async.Select
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+
+/--
+Unix style signals for Unix and Windows. SIGKILL and SIGSTOP are missing because they cannot be caught.
+SIGPIPE is not present because it's usually handled by the runtime.
+-/
+inductive Signal
+
+  /--
+  Hangup detected on controlling terminal or death of controlling process. SIGHUP is not
+  generated when terminal raw mode is enabled.
+
+  On Windows:
+  * SIGHUP is generated when the user closes the console window. The program is given ~10 seconds to
+    cleanup before Windows unconditionally terminates it.
+  -/
+  | sighup
+
+  /--
+  Interrupt program.
+
+  * Normally delivered when the user presses CTRL+C. Not generated when terminal raw mode is enabled (like Unix).
+  -/
+  | sigint
+
+  /--
+  Quit program.
+  -/
+  | sigquit
+
+  /--
+  Trace/breakpoint trap.
+  -/
+  | sigtrap
+
+  /--
+  Abort signal.
+
+  Notes:
+  - SIGABRT will not be caught by libuv if generated by abort(), e.g. via assert().
```

Comment:
I don't think users should care about libuv here; the docstring should just discuss the behavior from a user's perspective. Same for `sigwinch` below.